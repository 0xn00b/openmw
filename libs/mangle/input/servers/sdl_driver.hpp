#ifndef MANGLE_INPUT_SDL_DRIVER_H
#define MANGLE_INPUT_SDL_DRIVER_H

#include "../driver.hpp"

namespace Mangle
{
  namespace Input
  {
    /** Input driver for SDL. As the input system of SDL is seldomly
        used alone (most often along with the video system), it is
        assumed that you do your own initialization and cleanup of SDL
        before and after using this driver.

        The Event.event() calls will be given the proper EV_ type, the
        key index (for key up/down events), and a pointer to the full
        SDL_Event structure.
     */
    struct SDLDriver : Driver
    {
      void capture();
      bool isDown(int index);
      void showMouse(bool);
    };
  }
}
#endif
