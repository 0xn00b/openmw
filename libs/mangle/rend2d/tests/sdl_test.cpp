#include <iostream>
#include <fstream>

using namespace std;

#include "../servers/sdl_driver.hpp"

using namespace Mangle::Rend2D;

int main()
{
  cout << "Loading SDL driver.\n";
  SDLDriver sdl;

  cout << "Creating window.\n";
  sdl.setVideoMode(640,480);
  cout << "Current mode: " << sdl.width() << "x" << sdl.height() << endl;

  cout << "Setting fancy title, cause we like fancy titles.\n";
  sdl.setWindowTitle("Chief executive window");

  // Display surface
  Sprite *screen = sdl.getScreen();

  const char* imgName = "tile1-blue.png";
  cout << "Loading " << imgName << " from file.\n";
  Sprite *image = sdl.loadImage(imgName);

  const char* imgName2 = "tile1-yellow.png";
  cout << "Loading " << imgName2 << " from memory.\n";
  Sprite *image2;
  {
    // This is hard-coded for file sizes below 500 bytes, so obviously
    // you shouldn't mess with the image files.
    ifstream file(imgName2, ios::binary);
    char buf[500];
    file.read(buf, 500);
    int size = file.gcount();
    image2 = sdl.loadImage(buf, size);
  }

  cout << "Going bananas.\n";
  for(int i=1; i<20; i++)
    screen->draw(image, 30*i, 20*i);

  cout << "Taking a breather.\n";
  sdl.update();
  for(int i=1; i<20; i++)
    screen->draw(image2, 30*(20-i), 20*i);
  sdl.sleep(800);
  sdl.update();
  cout << "WOW DID YOU SEE THAT!?\n";
  sdl.sleep(800);

  cout << "Mucking about with the gamma settings\n";
  sdl.setGamma(2.0, 0.1, 0.8);
  sdl.sleep(100);
  sdl.setGamma(0.6, 2.1, 2.1);
  sdl.sleep(100);
  sdl.setGamma(1.6);
  sdl.sleep(100);

  cout << "Done.\n";
  return 0;
}
