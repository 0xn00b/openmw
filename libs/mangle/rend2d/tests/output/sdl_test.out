Loading SDL driver.
Creating window.
Current mode: 640x480
Setting fancy title, cause we like fancy titles.
Loading tile1-blue.png from file.
Loading tile1-yellow.png from memory.
Going bananas.
Taking a breather.
WOW DID YOU SEE THAT!?
Mucking about with the gamma settings
Done.
